### PR TITLE
tests(db/declarative): bootstrap correctly to avoid corrupted table

### PR DESCRIPTION
**Pull request by chronolaw**
### Summary

See KAG-4440 for details.

### Checklist

- [ ] The Pull Request has tests
- [x] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

Fix #_[issue number]_

### CODE REVIEW

File: spec/02-integration/03-db/08-declarative_spec.lua [hunk 1]
```diff
@@ -10,6 +10,11 @@ for _, strategy in helpers.each_strategy() do
     lazy_setup(function()
       local _
       _, db = helpers.get_db_utils(strategy)
+
+      -- ensure bootstrap DB correctly
```
Comment (StarlightIbuki) [inline spec/02-integration/03-db/08-declarative_spec.lua:14]: We need a longer comment to clarify the reason.

Comment (StarlightIbuki) [inline spec/02-integration/03-db/08-declarative_spec.lua:14]:
```suggestion
      -- This is a special case, where some DB states could be corrupted by DB truncation in `lazy_teardown()`.
      -- We manually bootstrap the DB here to ensure the creation of a table is done correctly
```

**Review by StarlightIbuki — CHANGES_REQUESTED** [review]

**Review by StarlightIbuki — APPROVED** [review]

File: spec/02-integration/03-db/08-declarative_spec.lua [hunk 2]
```diff
@@ -10,6 +10,11 @@ for _, strategy in helpers.each_strategy() do
     lazy_setup(function()
       local _
       _, db = helpers.get_db_utils(strategy)
```
Comment (Water-Melon) [inline spec/02-integration/03-db/08-declarative_spec.lua:12]: Can we simply set the parameter`skip_migrations` to `true` to bootstrap the database?

https://github.com/Kong/kong-ee/blob/master/spec/helpers.lua#L583

File: spec/02-integration/03-db/08-declarative_spec.lua [hunk 3]
```diff
@@ -10,6 +10,12 @@ for _, strategy in helpers.each_strategy() do
...
+
+      -- This is a special case, where some DB states could be corrupted by DB truncation in `lazy_teardown()`.
+      -- We manually bootstrap the DB here to ensure the creation of a table is done correctly
+      db:schema_reset()
```
Comment (Water-Melon) [inline spec/02-integration/03-db/08-declarative_spec.lua:16]: Perhaps we can place `db:schema_reset()` before `get_db_utils` and also remove ` helpers.bootstrap_database(db)`.
bootstrap_database will be called in `get_db_utils`. https://github.com/Kong/kong-ee/blob/master/spec/helpers.lua#L584

Comment (chronolaw) [reply spec/02-integration/03-db/08-declarative_spec.lua:16]: No, we must call `get_db_utils` first then get variable `db` to call other apis.

**Review by Water-Melon — APPROVED** [review]

**Pull request merged**